### ISSUE TITLE
Add F-Droid for installing apps 

### ISSUE THREAD

**Issue by Porrumentzio** [finnmglas/Launcher#72]
1. Go to `Settings>Apps>Install apps`
2. It will propose Browser and Google Play to install apps

It would be great if F-Droid would also be an option.
______
Also in `Settings>Meta`, there appears the Google Play logo, instead of a general App Installing Repo. If F-Droid installed, it offers both Google Play and F-Droid.

So changing that icon would be greatful, although I don't have a concrete proposal.

**Comment by finnmglas** [comment]
That's two valid points there:

1. `Settings>Apps>Install apps`: True, I did not notice that until you mentioned... I just found the reason:

See [line 60 of SettingsFragmentActions.kt](https://github.com/finnmglas/Launcher/blob/92c20c62d46656fd3afb8f0fdd144448798718b9/app/src/main/java/com/finnmglas/launcher/settings/actions/SettingsFragmentActions.kt#L60):

`Uri.parse("https://play.google.com/store/apps/"))` - the PlayStore url is opened using an `Intent` rather than a  more general Uri like `market://` which would include other stores.

2. `Settings>Meta`: If anyone can come up with a **font-awesome icon to replace the PlayStore icon**, just comment below or create a pull request with the new icon inside.

I can't think of any trivial icon to replace the current one with though...

**Comment by Porrumentzio** [comment]
Great!
About the icon, maybe a "download icon" would be enough, a downwards arrow.

What do you think? Could you add the "help needed" label?

**Comment by finnmglas** [comment]
I found a [store icon](https://fontawesome.com/icons/store?style=solid) that seems appropriate - we should use that... I liked the download icon idea but as the user already downloaded the app when he is in the in-app settings (obviously), the store icon is a reasonable choice :+1:

**Comment by finnmglas** [comment]
The meta settings ended up looking like this:

![Screenshot_20201018-234149_Launcher.jpg](https://user-images.githubusercontent.com/40366035/96386547-10d69200-119c-11eb-86d8-a1a7599335f6.jpg)

**Comment by finnmglas** [comment]
I created a [thread on StackOverflow](https://stackoverflow.com/q/64419005/12787264) about this topic. The official F-Droid application does not seem to listen to `Intent.CATEGORY_APP_MARKET` (not declared in its manifest), which makes fixing this issue even harder.

I received the suggestion that I could fix this by contributing to the F-Droid application... that kinda lays beyond the scope of Launcher though (but I actually consider doing it...)

**Comment by Porrumentzio** [comment]
Hi! It's a good icon, although I wouldn't understand it clearly because for me sofware is not for shopping, but yes for many people.

About the F-Droid intent, did you think about posting that problem into https://forum.f-droid.org?

**Comment by rptb1** [comment]
When DAVx5 needs to subscribe you to a WebCAL it asks you to install ICSx5, and it manages to do this via F-Droid.  I'm not sure if this is helpful, but the line of code is here https://gitlab.com/bitfireAT/davx5-ose/-/blob/dev-3.x-ose/app/src/main/java/at/bitfire/davdroid/ui/account/WebcalFragment.kt#L133 . This looks like a market intent that you're saying doens't work, but I'm pretty sure it popped up F-Droid on my phone.  However, I do have the F-Droid privileged extension installed, which may have something to do with it.

**Comment by rptb1** [comment]
Another possibly relevant thing. If I enter `market://details?id=at.bitfire.icsdroid` into the Firefox address bar and choose "open in app" I'm offered all possible app stores, so there's clearly a sort of "throw this out there" operation for such a URL.  This is F-Droid 1.10 on /e/OS (Android 10).

**Comment by finnmglas** [comment]
@rptb1 Yeah, that also works with Launcher - the `onClick` of the [fa-store](https://fontawesome.com/icons/store?style=solid) icon uses a `market://` intent that is received by F-droid, the PlayStore and the Samsung Galaxy Store - the only problem here was that I didn't manage to get a market uri to the market homepage - only to search or details pages ...

**Comment by toby-bro** [comment]
Isn't it possible to open the app directly by knowing that it is installed as "org.fdroid.fdroid" ? 
Otherwise couldn't it be possible to pass it as a spinner in the settings::launcher menu : default app repo ? By doing so the user can choose what the button is to do by default. Furthermore that would make it easy to implement it on the app.

**Comment by Kj-is-escape** [comment]
I think this issue is solved already, maybe by an update on f-droid. But in my phone i'm asked to select an app.

![Screenshot_20220814-162452_1](https://user-images.githubusercontent.com/68979736/184551889-1bf0331e-79a5-458b-b5fe-7f1c7fcc4104.png)
